### PR TITLE
Add devise

### PR DESCRIPTION
**Pull request by sja-thedude**
- Removed current_user method in ApplicationController.
- Install and set up devise.
Registers a new user.
User logs in with a combination of email and password.
Hashed passwords should be stored in the database.
Asks for confirmation of email.
Resets password.

Please Note: Though we confirmed user email, we are not sending actual confirmations to the mail. To confirm a new user mail, please check for the link provided in the local server.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,5 +1,18 @@
...
+  def update_allowed_parameters
+    devise_parameter_sanitizer.permit(:sign_up) { |u| u.permit(:name, :surname, :email, :password) }
+    devise_parameter_sanitizer.permit(:account_update) do |u|
+      u.permit(:name, :surname, :email, :password, :current_password)
+    end
   end
+
+  # def current_user
+  #   User.first
+  # end
```
Comment (andersonlebon) [inline app/controllers/application_controller.rb]: - [x] Kindly remove this commented code here to make your code clear.

File: app/views/devise/registrations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    <em>(<%= @minimum_password_length %> characters minimum)</em>
+    <% end %><br />
+    <%= f.password_field :password, autocomplete: "new-password" %>
+  </div>
+
+  <div class="field">
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
```
Comment (andersonlebon) [inline app/views/devise/registrations/new.html.erb]: - [x] Kindly check the wireframe and improve your styling. 

image

**Review by andersonlebon — CHANGES_REQUESTED** [review]
## Hi @mckent05 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there
## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
## Required Changes ♻️

- [ ] Kindly check comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by dasileker — APPROVED** [review]
### **Hi @sja-thedude,** 

### **Congratulations 👏👏**

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**